src/rpc/cli.cpp

```diff
@@ -216,6 +257,47 @@ static int cli_check_secret(const char *source)
    return 0;
 }
 
+static int cli_quitting = false;
```

Comment:
Global variables should usually be avoided.
I don't see how this could be done here, given the C-style interfacing of editline. Plus, editline itself does it under the hood as well.

At the very least, please set it to false in the `cli` constructor to allow cli restarts, should we ever want to implement them.

Reply:
I've modified some code in the direction of making CLI restart-able.